src/steps/device-groups/index.ts

```diff
@@ -0,0 +1,91 @@
+import {
+  createDirectRelationship,
+  Entity,
+  IntegrationStep,
+  IntegrationStepExecutionContext,
+  RelationshipClass,
+} from '@jupiterone/integration-sdk-core';
+
+import { createAPIClient } from '../../client';
+import { IntegrationConfig } from '../../config';
+import {
+  ACCOUNT_ENTITY_KEY,
+  Entities,
+  Relationships,
+  Steps,
+} from '../constants';
+import { createDeviceGroupEntity } from './converter';
+
+export async function fetchDeviceGroups({
+  instance,
+  jobState,
+}: IntegrationStepExecutionContext<IntegrationConfig>) {
+  const apiClient = createAPIClient(instance.config);
+
+  await apiClient.iterateDeviceGroups(async (deviceGroup) => {
+    const deviceGroupEntity = createDeviceGroupEntity(deviceGroup);
+
+    if (deviceGroupEntity) {
+      await jobState.addEntity(deviceGroupEntity);
+
+      await jobState.addRelationship(
+        createDirectRelationship({
+          _class: RelationshipClass.HAS,
+          from: (await jobState.getData(ACCOUNT_ENTITY_KEY)) as Entity,
```

Comment:
Please move this up on top somewhere (as described in other comments) :slightly_smiling_face: